File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt user for positive integer
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);

    // print an n-by-n grid of bricks/spaces
    for (int i = 1; i <= n; i++)
    {
        for (int j = 0; j < (n * 2); j++)
        {
            // if middle, add gap
            if (j == n)
            {
                printf("  ");
            }

            // Decide whether " ", or "#" based on location in grid
            if (j <= (n - i - 1))
            {
                printf(" ");
            }
            else if (j >= (n + i))
            {
                printf("");
            }
            else
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
def get_positive_int(prompt):
    while True:
        try:
            n = int(input(prompt))
            if 1 <= n <= 8:
                return n
            else:
                print("Please enter a positive integer between 1 and 8, inclusive.")
        except ValueError:
            print("Please enter a valid integer.")


def main():
    # Prompt user for positive integer between 1 and 8
    height = get_positive_int("Height: ")

    # Print the half-pyramids
    for i in range(1, height + 1):
        # Print spaces for the left pyramid
        for j in range(height - i):
            print(" ", end="")

        # Print hashes for the left pyramid
        for j in range(i):
            print("#", end="")

        # Print the gap (2 spaces)
        print("  ", end="")

        # Print hashes for the right pyramid
        for j in range(i):
            print("#", end="")

        # Move to the next line
        print()


if __name__ == "__main__":
    main()
